### PR TITLE
HHH 9966 mprove schema tooling support for creating catalogs and schemas

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-9966

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java [hunk 1]
```diff
@@ -1980,6 +1980,46 @@ public final String quote(String name) {
...
+	}
+
+	/**
+	 * Get the SQL command used to drop the named catalog
+	 *
+	 * @param catalogName The name of the catalog to be dropped.
+	 *
+	 * @return The drop command
+	 */
+	public String getDropCatalogCommand(String catalogName) {
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java]: Hmm.. I was thinking getDropCatalogCommand would reflexively return an array too.  Any particular reason you decided to keep just single String return here?

Also, what about creating/dropping schemas?  Shouldn't those also deal in arrays?

Comment (dreab8) [inline hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java]: I think It's a good idea to return an array also for getDropCatalogCommand and for the schemas, i'm going to change the code. Thanks

File: hibernate-core/src/main/java/org/hibernate/tool/schema/internal/SchemaCreatorImpl.java [hunk 2]
```diff
@@ -122,19 +123,30 @@ public void release() {
...
 				metadata.getDatabase().getJdbcEnvironment().getDialect(),
 				targets
 		);
 	}
 
 	@Override
-	public void doCreation(Metadata metadata, boolean createSchemas, Dialect dialect, Target... targets)
+	public void doCreation(Metadata metadata, boolean createNamespaces, Dialect dialect, Target... targets)
 			throws SchemaManagementException {
+		boolean tryToCreateCatalogs = false;
```
Comment (sebersole) [inline hibernate-core/src/main/java/org/hibernate/tool/schema/internal/SchemaCreatorImpl.java]: I wonder if it makes sense to move the creation and dropping of namespaces into separate methods?

**Comment by sebersole** [comment]
Other than that, this looks good.  Good job!

**Comment by dreab8** [comment]
PR mergerd

**Pull request closed without merging**